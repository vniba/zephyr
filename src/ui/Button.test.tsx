import { render, screen } from '@testing-library/react';
import Button from './Button.tsx';
import { expect } from 'vitest';

describe('Button component', function () {
  it('should render with correct text => Click', function () {
    render(<Button>Click</Button>);
    expect(screen.getByRole('button', { name: /click/i })).toBeInTheDocument();
  });

  it('should matches default button snapshot', function () {
    const { container } = render(<Button>test</Button>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
