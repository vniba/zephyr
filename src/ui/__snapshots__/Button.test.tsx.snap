// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > should matches default button snapshot 1`] = `
.c0 {
  border: none;
  box-shadow: var(--shadow-sm);
  border-radius: var(--border-radius-md);
  font-size: 1.4rem;
  padding: 1.2rem 1.6rem;
  font-weight: 500;
  color: var(--color-brand-50);
  background-color: var(--color-brand-600);
}

.c0:hover {
  background-color: var(--color-brand-700);
}

<button
  class="c0"
  variation="primary"
>
  test
</button>
`;
